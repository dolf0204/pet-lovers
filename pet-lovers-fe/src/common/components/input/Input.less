.input__label {
    font-family: var(--ff-secondary);
    font-size: 0.938rem;
    color: var(--color-green-primary);
}

.input__input {
    height: 40px;
}

.input__input:hover {
    border-color: var(--color-green-primary);
}

.input__input:focus {
    border-color: var(--color-green-primary);
    box-shadow: none;
}
.is-required:after {
    content: ' *';
    color: var(--color-required);
}
