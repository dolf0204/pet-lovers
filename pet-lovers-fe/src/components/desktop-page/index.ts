import { lazy } from "react";

export const DesktopPage = lazy(() => import("./DesktopPage"));
