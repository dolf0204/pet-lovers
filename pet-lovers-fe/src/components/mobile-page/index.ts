import { lazy } from "react";

export const MobilePage = lazy(() => import("./MobilePage"));
