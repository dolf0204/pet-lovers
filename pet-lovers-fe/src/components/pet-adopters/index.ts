import { lazy } from "react";

export const PetAdopters = lazy(() => import("./PetAdopters"));
