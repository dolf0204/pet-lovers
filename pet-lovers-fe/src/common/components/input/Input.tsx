import { ChangeEvent, FC } from "react";

import { Input as AntInput } from "antd";

import "./Input.less";

interface IDefaultProps {
  label?: string;
  name?: string;
  onChange?: (e: ChangeEvent<HTMLInputElement>) => void;
  disabled?: boolean;
  required?: boolean;
}

type TProps = IDefaultProps;

export const Input: FC<TProps> = (props) => {
  const { label, name, onChange, disabled, required } = props;

  return (
    <div className="input">
      {label && (
        <label
          htmlFor={name}
          className={`input__label
                     ${!!required ? "is-required" : ""}`}
        >
          {label}
        </label>
      )}
      <AntInput
        disabled={disabled}
        onChange={onChange}
        name={name}
        className="input__input"
        onWheelCapture={(e) => e.currentTarget.blur()}
      />
    </div>
  );
};
