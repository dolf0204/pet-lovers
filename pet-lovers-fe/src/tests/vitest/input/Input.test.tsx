import { describe, expect, it } from "vitest";
import { act } from "@testing-library/react";
import { INPUT_COMPONENT_TEST_ID } from "./testIds";
import { Input } from "../../../common/components/input";
import { render, screen, userEvent } from "../utils/testUtils";

describe("Input component", () => {
  describe("mounting, and snapshot", () => {
    it("should render when mounted in the DOM tree", () => {
      render(<Input testId={INPUT_COMPONENT_TEST_ID} />);
      const input = screen.getByTestId(INPUT_COMPONENT_TEST_ID);
      expect(input).toBeInTheDocument();
    });

    it("should match snapshot", () => {
      render(<Input testId={INPUT_COMPONENT_TEST_ID} />);
      const input = screen.getByTestId(INPUT_COMPONENT_TEST_ID);
      expect(input).toMatchSnapshot();
    });
  });

  describe("type prop", () => {
    it("should render with type='text' when type prop is not passed", () => {
      render(<Input testId={INPUT_COMPONENT_TEST_ID} />);
      const input = screen.getByTestId(INPUT_COMPONENT_TEST_ID);
      expect(input).toHaveAttribute("type", "text");
    });
  });

  describe("disabled prop", () => {
    it("should render with disabled when disabled prop is passed", () => {
      render(<Input testId={INPUT_COMPONENT_TEST_ID} disabled />);
      const input = screen.getByTestId(INPUT_COMPONENT_TEST_ID);
      expect(input).toBeDisabled();
    });
  });

  describe("testing label", () => {
    it("should render with label when label prop is passed", () => {
      render(<Input testId={INPUT_COMPONENT_TEST_ID} label="test" />);
      const label = screen.getByText("test");
      expect(label).toBeInTheDocument();
    });

    it("should render htmlFor attribute with name prop when label and name props are passed", () => {
      render(
        <Input testId={INPUT_COMPONENT_TEST_ID} label="test" name="name" />
      );
      const label = screen.getByText("test");
      expect(label).toHaveAttribute("for", "name");
    });
  });

  describe("handle input change", () => {
    it("should call onChange when input value changes", async () => {
      const user = userEvent.setup();
      const onChange = vi.fn();
      render(<Input testId={INPUT_COMPONENT_TEST_ID} onChange={onChange} />);
      const input = screen.getByTestId(INPUT_COMPONENT_TEST_ID);
      await act(async () => {
        await user.type(input, "test");
      });
      expect(onChange).toHaveBeenCalled();
      expect(input.getAttribute("value")).toBe("test");
    });

    it("should allow string input when type='text'", async () => {
      const user = userEvent.setup();
      render(<Input testId={INPUT_COMPONENT_TEST_ID} />);
      const input = screen.getByTestId(INPUT_COMPONENT_TEST_ID);
      await act(async () => {
        await user.type(input, "123test");
      });
      expect(input.getAttribute("value")).toBe("123test");
    });
  });
});
