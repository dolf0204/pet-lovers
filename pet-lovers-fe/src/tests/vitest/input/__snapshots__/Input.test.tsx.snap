// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Input component > mounting, and snapshot > should match snapshot 1`] = `
<input
  class="ant-input css-dev-only-do-not-override-yp8pcc input__input"
  data-testid="input-test-id"
  type="text"
  value=""
/>
`;
